lib/matplotlib/__init__.py

```diff
@@ -1296,15 +1296,37 @@ def use(backend, *, force=True):
     rcParams['backend'] = os.environ.get('MPLBACKEND')
 
 
-def get_backend():
+def get_backend(*, resolve=True):
     """
     Return the name of the current backend.
 
+    Parameters
+    ----------
+    resolve : bool, default: True
+        Whether to trigger backend resolution if no backend has been
+        selected so far. If True, this ensures that a valid backend
+        is returned. If False, this returns None if no backend has been
+        selected so far.
+
+        .. admonition:: Provisional resolve flag
+
+           The *resolve* flag is introduced provisionally in the anticipation
+           of backend resolution refactoring. We don't guarantee API stability
+           for now, but if all plays out well, the provisional availability
+           will prolong the range of supported releases for the migration period.
+
```

Comment:
This is vague and confusing.  Either there is a plan, in which case explain, or just state the possible issues a user may experience here?  

I'm also not clear why this is "provisional" at all? Do you mean it may go away again?  Why would that happen?

Reply:
I've removed the motivation.

As written, this anticipates a planned refactoring of the backend mechanism https://github.com/matplotlib/matplotlib/issues/26406#issuecomment-2442816302. I'm 95%, we'll have this API in the end, but until the whole refactoring is done, there may be some aspects we've overlooked that would make it necessary or desirable to still change this. I've made this provisional to be able to easily react to this 5% change of change.